### PR TITLE
Support saving the generated code

### PR DESCRIPTION
**Pull request by ken-matsui**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -197,7 +203,22 @@ where
...
+
+    // Save the generated code.
+    if let Some(save_code_dir) = &args.save_code_dir {
+        // Create the directory if it doesn't exist.
+        create_dir_all(save_code_dir)?;
+
+        let mut code_path = save_code_dir.clone();
+        code_path.push("main.c");
+        let code_file = File::create(&code_path)?;
+        // TODO: Emitting the code twice is not ideal.
```
Comment (ken-matsui) [inline src/main.rs:215]: This TODO can be resolved by https://github.com/samkaufman/morello/pull/20

**Comment by samkaufman** [comment]
@ken-matsui What's the use case for this PR?

**Comment by samkaufman** [comment]
PR #24 may be relevant to the intended use case.

**Comment by ken-matsui** [comment]
With the patch, we can use pipe. This PR would be unnecessary.

**Pull request closed without merging**